Dev/add user logout

my8bird opened the pull request:
Adds a new Auth service that handles the authentication and storage of the token (keeps the `gh` service light).

Allows users to logout

Cleans up `gh` service a bit

my8bird commented on src/app/app.js:
```diff
@@ -4,6 +4,7 @@ angular.module( 'Trestle', [
    'Trestle.board',
    'Trestle.login',
    'Trestle.issue',
+   'Trestle.authentication',
```
We can fix this latter if we want I was not sure how that last conversation ended up

my8bird commented on src/app/board/toolbar/toolbar.scss:
```diff
@@ -3,7 +3,7 @@
     }
 
-    .dropdown-menu {
+    #repo-list {
```
update to be more specific so that not all menu's get this styling

my8bird commented on src/app/board/toolbar/toolbar.scss:
```diff
@@ -26,4 +26,10 @@
         margin: 5px;
     }
+
+    #settings {
```
Is there a better name for the area that holds the users avatar, name and the settings gear?

my8bird commented on src/app/board/toolbar/toolbar.tpl.html:
```diff
@@ -36,15 +36,32 @@
 
-       <li class="pull-right">
+      <li>
         <a ng-href="#board/{{toolbarCtrl.owner}}/{{toolbarCtrl.repo}}">Board</a>
```
These appear in the wrong order but the `float: right`'s cause them to render correctly

my8bird commented on src/app/board/toolbar/toolbar.tpl.html:
```diff
@@ -36,15 +36,32 @@
+        <div class="dropdown-menu">
+          <span ng-click="userCtrl.onLogout()">Logout</span>
+        </div>
+      </li>
```
this could be a directive but I thought i would hold off until we get done feeling that out the issue cards

my8bird commented on src/app/login/services.js:
```diff
@@ -0,0 +1,91 @@
+   };
+
+   this._onLoginError = function(res) {
+      console.warn('login error');
```
Really nothing to do this the promise will pass everything back.  We could extract data from the failed HTTP request and return it in a different format (so that callers do not need to handle the HTTP stuff ) but that seems like another pull (for showing the login error)

my8bird commented on src/common/github_api/github_api.js:
```diff
@@ -22,22 +22,7 @@ angular.module('github.api', ['restangular'])
  Angular service `gh` which provides tools for accessing GitHub API's
  */
 .service('gh', function gh(GitHubRestangular, $http, $interpolate, $rootScope, $q) {
```
The changes in this file simply kill code that was moved.

my8bird commented on src/common/github_api/github_api.spec.js:
```diff
@@ -16,40 +16,6 @@ describe( 'GitHub API (gh)', function() {
       expect( gh.hasAccessToken() ).not.toBeTruthy();
    }));
 
-   describe( 'setAccessToken', function() {
```
these tests no longer exist since the `gh` service does not have this interface anymore

my8bird commented:
@abierbaum still looking for reviews on this one

abierbaum commented on src/app/board/main/main.js:
```diff
@@ -1,10 +1,18 @@
 angular.module('Trestle.board')
 
-.controller( 'BoardCtrl', function($stateParams, $location, gh) {
+.controller( 'BoardCtrl', function($stateParams, $location, auth, gh) {
```
Probably needs to be on repoctrl in new world order.

abierbaum commented on src/app/login/services.js:
```diff
@@ -0,0 +1,193 @@
+ Provides tools for logging in and out of the system and presisting the
+ authentication across browser refreshes.
+ */
+.service('auth', function($http) {
```
Move to services/auth_service.js or something like that.

my8bird replied:
done

abierbaum commented on src/app/login/services.js:
```diff
@@ -0,0 +1,193 @@
+    */
+   this.logout = function() {
+      // Clear the in memory version
+      token = null;
```
double check that github service has it's value cleared as well.

my8bird replied:
done

abierbaum commented on src/app/login/services.js:
```diff
@@ -0,0 +1,193 @@
+ @name  Trestle.authentication
+
+ @description
+ Holds services, controllers, etc related to authenticating users with the system.
```
Maybe a good idea to describe what we are doing.

abierbaum commented on src/app/login/login.js:
```diff
@@ -37,48 +34,26 @@ angular.module( 'Trestle.login', [
-
+         },
+         function() {
+            console.log('more error');
```
It would be good to set an alert bar.

abierbaum commented:
+1

Commit:
Add an service which tracks authentication for the system

Commit:
Add ability to get the logged in users details

Also, strip out code that was used to track the logged in user.  This is now the responsibility of the user of the library

Commit:
feature: Show logged in user and allow them to logout
